src/block/Stair.php

```diff
@@ -128,4 +129,11 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 
 		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 	}
+
+	public function getSupportType(int $facing) : SupportType{
+		if($facing === Facing::UP && $this->isUpsideDown() || $facing === $this->facing || $facing === Facing::DOWN && !$this->isUpsideDown()){
+			return SupportType::FULL();
```

Comment:
This needs to include logic for corner stairs.

Reply:
This is also potentially subject to precedence issues (needs more ().)

Reply:
in fact, the corner stairs are in this case problematic

Reply:
Do we really implement as vanilla or do we add this feature: 
```
Take a corner stair type of INNER_RIGHT facing north.
In vanilla, you will be able to put torch on the north face and not on the east.
Knowing that east face are full, should we allowed the torch placement or not ?
```